scalameta/transversers/shared/src/test/scala/meta/transversers/OrderTests.scala

```diff
@@ -0,0 +1,109 @@
+package scala.meta.transversers
+
+import scala.meta._
+import org.scalatest._
+import scala.meta.quasiquotes._
+import scala.collection.mutable
+
+/**
+ * Note: Traversal is not exactly pre/post/in order.
+ * The children aare sorted. For most cases we only
+ * care about leaf first or parent first
+ */
+class OrderTests extends FunSuite {
+
+  val rootFirst = mutable.ArrayBuffer(
+    "scala.meta.Defn$Class$DefnClassImpl",
+    "scala.meta.Type$Name$TypeNameImpl",
+    "scala.meta.Ctor$Primary$CtorPrimaryImpl",
+    "scala.meta.Name$Anonymous$NameAnonymousImpl",
+    "scala.meta.Template$TemplateImpl",
+    "scala.meta.Self$SelfImpl",
+    "scala.meta.Name$Anonymous$NameAnonymousImpl",
+    "scala.meta.Defn$Def$DefnDefImpl",
+    "scala.meta.Term$Name$TermNameImpl",
+    "scala.meta.Lit$Int$LitIntImpl")
+
+    val leafFirst = mutable.ArrayBuffer(
+      "scala.meta.Type$Name$TypeNameImpl",
+      "scala.meta.Name$Anonymous$NameAnonymousImpl",
+      "scala.meta.Ctor$Primary$CtorPrimaryImpl",
+      "scala.meta.Name$Anonymous$NameAnonymousImpl",
+      "scala.meta.Self$SelfImpl",
+      "scala.meta.Term$Name$TermNameImpl",
+      "scala.meta.Lit$Int$LitIntImpl",
+      "scala.meta.Defn$Def$DefnDefImpl",
+      "scala.meta.Template$TemplateImpl",
+      "scala.meta.Defn$Class$DefnClassImpl")
+
+  val input = q"class Foo { def bar = 1 }"
+
+  test("Traverse visits roots first") {
+    val visited: mutable.ArrayBuffer[String] = mutable.ArrayBuffer()
+
+    input.traverse {
+      case a => visited.append(a.getClass.getName)
```

Comment:
FYI `.productPrefix` will give you a cleaner name per tree node

Reply:
Nice! Usually I use getClass.getSimpleName but of course that fails here.